server/controllers/authentication.js

```diff
@@ -0,0 +1,35 @@
+import User from '../models/userModel';
+import { buildResponse } from '../utils/helpers';
+
+exports.signUp = (req, res, next) => {
+  const { name, email, password } = req.body;
+  if (!email || !password) {
+    const errorMessage = 'You must provide email and password';
+    const response = buildResponse(false, errorMessage);
+    return res.status(400).send(response);
+  }
+  User.findOne({ email }, (err, existingUser) => {
```

Comment:
why not async/await?

Reply:
Refactored using async/ await. Found it much better than callbacks.